src/main/java/frc/robot/Robot.java

```diff
@@ -1,12 +1,38 @@
 package frc.robot;
 
+import edu.wpi.first.networktables.NetworkTableInstance;
 import edu.wpi.first.wpilibj.TimedRobot;
 import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import frc.robot.subsystems.vision.coordinate.CoordinateDistance;
+import frc.robot.subsystems.vision.provider.CommonVisionProvider;
+import frc.robot.subsystems.vision.provider.Vision;
+
 
 public class Robot extends TimedRobot {
 
     @Override
     public void robotInit() {
+
+        NetworkTableInstance commonInstance = NetworkTableInstance.getDefault(); //in case anyone else requires networktables access
+        commonInstance.startClientTeam(4183);
+
+        Vision vision = new CommonVisionProvider(commonInstance, 0.0).initialize();
+        //do something with the vision """"subsystem"""" (if you need it dependency inject it into your subsystem)
+
+        if (vision.hasValidTarget()) {
+            System.out.println("target acquired - arming salvo");
+        }
+
+        vision.getCoordinatesInstant().ifPresent(coords -> {
+
+            System.out.println(coords.getHorizontalOffset());
+
+            double toTarget = new CoordinateDistance(coords).distance();
+
+            System.out.println(toTarget);
+        });
```

Comment:
This is just sample code on how to use the vision system, right? Because it will only execute once and we rarely have a target when the robot is in robotInit. I'm not sure the limelight will even be functioning at this point.

Reply:
yes, it is sample code.